octobot_trading/exchanges/util/parser/util.py

```diff
@@ -0,0 +1,291 @@
+                self._log_missing_with_method(
+                    key_to_set=key_to_set,
+                    tried_message=keys_to_test,
+                    method=not_found_method,
+                    error=e,
+                )
+        elif enable_log:
+            self._log_missing(
+                key_to_set=key_to_set,
+                tried_message=keys_to_test,
+            )
+        self.formatted_record[key_to_set] = not_found_val
+        return False
+
+    def _try_to_find_and_set_decimal(
+        self,
+        key_to_set,
+        keys_to_test,
+        not_found_val=0,
+        parse_method=None,
+        not_found_method=None,
+        enable_log=True,
+        use_dict_root=True,
+        use_info_sub_dict=False,
+    ) -> bool:
+        if value := decimal.Decimal(
+            str(
+                self._try_to_find(
+                    key_to_set=key_to_set,
+                    keys_to_test=keys_to_test,
+                    not_found_val=not_found_val,
+                    enable_log=False,
+                    use_dict_root=use_dict_root,
+                    use_info_sub_dict=use_info_sub_dict,
+                )
+            )
+        ):
+            self.formatted_record[key_to_set] = (
+                parse_method(value) if parse_method else value
+            )
+            return True
+
+        if not_found_method:
+            try:
+                self.formatted_record[key_to_set] = decimal.Decimal(str(not_found_method()))
+                return False
+            except Exception as e:
+                self._log_missing_with_method(
+                    key_to_set=key_to_set,
+                    tried_message=keys_to_test,
+                    method=not_found_method,
+                    error=e,
+                )
+                return False
+        if enable_log and value != trading_constants.ZERO:
+            self._log_missing(
+                key_to_set=key_to_set,
+                tried_message=keys_to_test,
+            )
+        self.formatted_record[key_to_set] = decimal.Decimal(str(not_found_val))
+        return False
+
+    def _try_to_find(
+        self,
+        key_to_set,
+        keys_to_test,
+        not_found_val=None,
+        enable_log=True,
+        use_dict_root=True,
+        use_info_sub_dict=False,
+    ):
+        value = None
+        for key in keys_to_test:
+            try:
+                if use_dict_root:
+                    value = self.raw_record[key]
+                    if (value != 0 and value is not None) or value is False:
+                        return value
+            except:
+                pass
+            try:
+                if use_info_sub_dict:
+                    value = self.raw_record[ExchangePositionCCXTColumns.INFO.value][key]
+                    if (value != 0 and value is not None) or False:
+                        return value
+            except:
+                pass
+        if value != float(0) and enable_log:
```

Comment:
```suggestion
        if value != 0.0 and enable_log:
```

Reply:
I improved handling of falsely values in the last two commits